File: 백준/Gold/3020. 개똥벌레/개똥벌레.java
import java.awt.List;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;
public class Main{
	public static void main (String [] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer nh = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(nh.nextToken());
		int H = Integer.parseInt(nh.nextToken());
		int [] arr = new int[H+1];
		int [] front = new int[H+1];
		int [] back = new int[H+1];
		
		for (int i = 0; i < N; i++) {
				StringTokenizer st = new StringTokenizer(br.readLine());
				int n = Integer.parseInt(st.nextToken());
				if (i% 2 != 0) {
					front[n] +=1;					
				}
				else {
					back[H - n +1] += 1;
				}
		}
		for (int i = H; i > 0; i--) {
			front[i-1] += front[i];
		}
		for (int i = 0; i < H; i++) {
			back[i+1] += back[i];
		}
		for (int i = 0; i < arr.length; i++) {
			arr[i] = front[i] + back[i];
		}
		int val = 200000;
		int cnt = 0;
		for (int i = 1; i < H+1; i++) {
			if (val > arr[i]) {
				val = arr[i];
				cnt = 1;
			}
			else if (val == arr[i]) {
				cnt += 1;
			}
		}
		System.out.print(val);
		System.out.print(" ");
		System.out.println(cnt);
	}
}

File: 백준/Gold/3020. 개똥벌레/개똥벌레.py
import sys
input = sys.stdin.readline
N, H = map(int, input().split())
arr = [0 for _ in range(H+1)]
front = [0 for _ in range(H+1)]
back = [0 for _ in range(H+1)]
for i in range(N):
    if i % 2:
        back[H - int(input()) + 1] += 1
    else:
        front[int(input())] += 1

for i in range(1, H+1):
    back[i] += back[i-1]

for i in range(H, 0, -1):
    front[i-1] += front[i]

for i in range(H+1):
    arr[i] = front[i] + back[i]

val = 200000
cnt = 0
for i in range(1, H+1):
    if arr[i] < val:
        val = arr[i]
        cnt = 1
    elif arr[i] == val:
        cnt += 1
        
print(val, cnt)